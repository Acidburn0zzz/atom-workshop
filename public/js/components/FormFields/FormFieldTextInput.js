import React from 'react';

export default class FormFieldTextInput extends React.Component {

  static propTypes = {
    fieldLabel: React.PropTypes.string,
    fieldName: React.PropTypes.string,
    fieldValue: React.PropTypes.string,
    fieldPlaceholder: React.PropTypes.string,
    onUpdateField: React.PropTypes.func,
    isValid: React.PropTypes.bool
  };

  onUpdate = (e) => {
    this.props.onUpdateField(e.target.value);
  }


  render() {
    return (
        <div>
          <label htmlFor={this.props.fieldName} className="form__label">{this.props.fieldLabel}</label>
          <input type="text" className="form__field" id={this.props.fieldName} placeholder={this.props.fieldPlaceholder || ''} value={this.props.fieldValue || ""} onChange={this.onUpdate}/>
        </div>

    );
  }
}
