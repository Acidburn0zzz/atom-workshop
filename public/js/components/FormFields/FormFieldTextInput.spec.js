import React from 'react';
import FormFieldTextInput from './FormFieldTextInput';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue= 'test';

test('Should render', () => {
  const component = renderer.create(
    <FormFieldTextInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const input = shallow(
    <FormFieldTextInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );

  input.find('input').simulate('change');

  expect(updateFn).toHaveBeenCalledTimes(1);
});
