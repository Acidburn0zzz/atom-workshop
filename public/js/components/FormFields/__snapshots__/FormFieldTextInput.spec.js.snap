exports[`test Should render 1`] = `
<div>
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <input
    className="form__field"
    id="test"
    onChange={undefined}
    placeholder=""
    type="text"
    value="test" />
</div>
`;
